From bd0267600ecfedd0abe3bc08f0299f1136e51d51 Mon Sep 17 00:00:00 2001
From: Tony LIU <junjie.liu@freescale.com>
Date: Tue, 17 Apr 2012 11:12:04 +0800
Subject: [PATCH 312/435] ENGR00179763 add usb unit test case

- add udisk as MSC storage case
- add system suspend/resume while connected to usb charger case

Signed-off-by: Tony LIU <junjie.liu@freescale.com>
---
 test/doc/ut_um.toc                       |  2 ++
 test/mxc_usb_test/doc/FSL-UT-USB-014.txt | 38 ++++++++++++++++++++++++++++
 test/mxc_usb_test/doc/FSL-UT-USB-015.txt | 43 ++++++++++++++++++++++++++++++++
 3 files changed, 83 insertions(+)
 create mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-014.txt
 create mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-015.txt

diff --git a/test/doc/ut_um.toc b/test/doc/ut_um.toc
index 24a3511..118acc6 100644
--- a/test/doc/ut_um.toc
+++ b/test/doc/ut_um.toc
@@ -133,6 +133,8 @@ include::../mxc_usb_test/doc/FSL-UT-USB-010.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-011.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-012.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-013.txt[]
+include::../mxc_usb_test/doc/FSL-UT-USB-014.txt[]
+include::../mxc_usb_test/doc/FSL-UT-USB-015.txt[]
 
 V4L2
 ----
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-014.txt b/test/mxc_usb_test/doc/FSL-UT-USB-014.txt
new file mode 100644
index 0000000..7d124cc
--- /dev/null
+++ b/test/mxc_usb_test/doc/FSL-UT-USB-014.txt
@@ -0,0 +1,38 @@
+FSL-UT-USB-014-UDISK-AS-MSC-Storage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Udisk as MSC storage
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+
+1. Attach a udisk to host1 port, and wait a device created, assuming this device
+is /dev/sdb1
+
+2. modprobe g_storage file=/dev/sdb1 removable=1 serial="123456789012"
+
+3. connect OTG port with PC, on PC side, one removable storage can be found
+
+4. read/write data from/to this removable storage and verify
+
+| Expected Result |
+Pass
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-015.txt b/test/mxc_usb_test/doc/FSL-UT-USB-015.txt
new file mode 100644
index 0000000..ee3d55e
--- /dev/null
+++ b/test/mxc_usb_test/doc/FSL-UT-USB-015.txt
@@ -0,0 +1,43 @@
+FSL-UT-USB-015-SUSPNED-RESUME-WHILE-OTG-CONNECTED-WITH-USB-CHARGER
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+system suspend/resume test while OTG port connected with usb charger
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+
+1. modprobe g_serial
+
+2. connect OTG port to USB charger
+
+3. echo enabled > /sys/devices/platform/imx-uart.0/tty/ttymxc0/power/wakeup
+
+4. echo mem > /sys/power/state
+
+5. after system suspended, resume system with UART port
+
+6. repeat step 4,5 for at least 5 times, no system hang should be found
+
+7. disconnect OTG port to USB charger, check USB clock, it should be turned off
+
+| Expected Result |
+Pass
+
+|====================================================================
+
+<<<
-- 
1.8.0

